pkg/log/checkpoint/connectivity.go

```diff
@@ -0,0 +1,34 @@
+package checkpoint
+
+import (
+	"fmt"
+	"net"
+)
+
+type dialer interface {
+	Dial(network, address string) (net.Conn, error)
+}
+
+func testConnections(dialer dialer, hosts ...string) []string {
+	results := []string{}
```

Comment:
I could roll with this. But, what about `results := map[string]string` and then let the eventual json marshaling collapse it?

Reply:
All of the funcs that are called by `logCheckPoint(logger log.Logger)` return a `[]string`. Are you suggesting that we convert them all to `map[string]string` where the key is either the hostname, url, or filepath? I like that ... looks like this in log
```
"IPs":
  {
    "dl.kolide.co":"35.186.193.229",
    "k2control.kolide.com":"18.205.222.128, 54.237.133.81, 54.161.241.46, 52.202.168.65",
    "k2device.kolide.com":"54.161.241.46, 18.205.222.128, 54.237.133.81, 52.202.168.65",
    "notary.kolide.co":"35.196.217.93"
  }
```

Reply:
I'm not sure they all need to return the same thing. The underlying logger interface is `keyvals ...interface{}`, and I'm pretty sure that under the hood that's getting converted to json. I think for something like hostname, we should return `string`, but for something like this it feels much more like `map[string]interface{}`, which lets us stick whatever we want in there. 

Of course, I'm assuming logger will marshal it, but I think so. And it seems cleaner than hand rolling what looks like json

Reply:
Hrm, really? https://go.dev/play/p/QiW45PhqbFy

Reply:
Okay, I see what I did here, I was passing just the map with out a key `logger.Log(someMap)` https://go.dev/play/p/CIiuyljG52A and it was giving me the funky output ... but `logger.Log("someKey", someMap)` works as expected.